contrib/check-semver.sh

```diff
@@ -0,0 +1,202 @@
+    generate_json_files_no_default_features "bitcoin_hashes" "current"
+    generate_json_files_features_alloc "bitcoin_hashes" "current"
+
+    # 4. bitcoin-units: no-default-features and alloc feature.
+    generate_json_files_no_default_features "bitcoin-units" "current"
+    generate_json_files_features_alloc "bitcoin-units" "current"
+
+    # 5. bitcoin-io: no-default-features and alloc feature.
+    generate_json_files_no_default_features "bitcoin-io" "current"
+    generate_json_files_features_alloc "bitcoin-io" "current"
+
+
+    # Switch to target commit.
+    echo "Checking out target commit at $TARGET_COMMIT"
+    git checkout "$TARGET_COMMIT"
+
+    # On target commit:
+    # 1. bitcoin: all-features and no-default-features.
+    generate_json_files_all_features "bitcoin" "master"
+    generate_json_files_no_default_features "bitcoin" "master"
+
+    # 2. base58ck: all-features and no-default-features.
+    generate_json_files_all_features "base58ck" "master"
+    generate_json_files_no_default_features "base58ck" "master"
+
+    # 3. bitcoin_hashes: no-default-features and alloc feature.
+    generate_json_files_no_default_features "bitcoin_hashes" "master"
+    generate_json_files_features_alloc "bitcoin_hashes" "master"
+
+    # 4. bitcoin-units: no-default-features and alloc feature.
+    generate_json_files_no_default_features "bitcoin-units" "master"
+    generate_json_files_features_alloc "bitcoin-units" "master"
+
+    # 5. bitcoin-io: no-default-features and alloc feature.
+    generate_json_files_no_default_features "bitcoin-io" "master"
+    generate_json_files_features_alloc "bitcoin-io" "master"
+
+    # Check for API semver breaks on all the generated JSON files above.
+    run_cargo_semver_check "bitcoin" "all-features"
+    run_cargo_semver_check "bitcoin" "no-default-features"
+    run_cargo_semver_check "base58ck" "all-features"
+    run_cargo_semver_check "base58ck" "no-default-features"
+    run_cargo_semver_check "bitcoin_hashes" "no-default-features"
+    run_cargo_semver_check "bitcoin_hashes" "alloc"
+    run_cargo_semver_check "bitcoin-units" "no-default-features"
+    run_cargo_semver_check "bitcoin-units" "alloc"
+    run_cargo_semver_check "bitcoin-io" "no-default-features"
+    run_cargo_semver_check "bitcoin-io" "alloc"
+
+    # Invoke cargo semver-checks to check for breaking changes
+    # in all generated files.
+    check_for_breaking_changes
+}
+
+# Run cargo doc with the cargo semver-checks rustdoc flags.
+# We don't care about dependencies.
+run_cargo_doc() {
+    RUSTDOCFLAGS="$RUSTDOCFLAGS" cargo +"$NIGHTLY" doc --no-deps "$@"
+}
+
+# Run cargo semver-check
+run_cargo_semver_check() {
+    local crate="$1"
+    local variant="$2"
+
+    echo "Running cargo semver-checks for $crate $variant"
+    # Hack to not fail on errors.
+    # This is necessary since cargo semver-checks will fail if the
+    # semver check fails.
+    # We check that manually later.
```

Comment:
There's a less brittle hack for that: append `|| true` at the end of the command.

Reply:
Yeah that was my initial implementation.